### DISCUSSION TITLE
GetSubstructMatches() fail to identify a match

### DISCUSSION THREAD

**Discussion by swu1992** [rdkit/rdkit#7558] [Q&A] [answered]
Dear all, 

I am trying to use GetSubstructMatches() to find the atoms in a molecule that match a fragment from a SMARTS string, but fail to return the match as I expected. Below are two examples. I would appreciate any comments. 

```
mol1 = Chem.MolFromSmiles('O=C(CCc1ccccc1)/N=c1\[n-]nc(-c2ccccc2I)s1')
mol2 = Chem.MolFromSmarts('*C(=O)/N=c1\[n-]nc(*)s1')
print(mol1.GetSubstructMatches(mol2))
```
![image](https://github.com/rdkit/rdkit/assets/20412574/b3bc4b61-a441-48db-a647-5b35ce23e3dc)

```
mol1 = Chem.MolFromSmiles('[H]/N=c1\[n-]nc(-c2ccccc2)o1')
mol2 = Chem.MolFromSmarts('*c1n[n-]/c(=N\[H])o1')
print(mol1.GetSubstructMatches(mol2))
```
![image](https://github.com/rdkit/rdkit/assets/20412574/93ee3594-3a65-4634-a822-e4207cfd8cb6)

**Comment by greglandrum** [comment, marked as answer]
@swu1992 what is happening here is that the `/` or `\` in your SMARTS are being interpreted as single bonds (and not indicating bond direction), so they do not match the aromatic bonds in your molecule. This is an old bug in the SMARTS parser


If you get rid of those markers things work as expected:
```
In [6]: mol1 = Chem.MolFromSmiles('O=C(CCc1ccccc1)/N=c1\[n-]nc(-c2ccccc2I)s1')
   ...: mol2 = Chem.MolFromSmarts('*C(=O)/N=c1[n-]nc(*)s1')
   ...: print(mol1.GetSubstructMatches(mol2))
((2, 1, 0, 10, 11, 12, 13, 14, 15, 22),)
```
